### ISSUE TITLE
Incompatible with BetterShulkers

### ISSUE THREAD

**Issue by arxenix** [MisterPeModder/ShulkerBoxTooltip#30]
Unfortunately I think ShulkerBoxTooltip is incompatible with my mod, BetterShulkers, because it assumes that the shulker box size is always 27. My mod customizes shulker boxes and increases their inventory size if they have certain enchantments.

It should be possible for you to retrieve the modified size by checking the max index of an item in the Inventory tag and rounding up to the next multiple of 9. If you could do that, it would be compatible.

My mod: https://www.curseforge.com/minecraft/mc-mods/fabric-bettershulkers

https://github.com/arxenix/better-shulkers

**Comment by MisterPeModder** [comment]
ShulkerBoxTooltip has an Plugin system for custom previews.
To use it, create a class that implements `ShulkerBoxTooltipApi` add it as an entrypoint in the fabric.mod.json as such:
```json
"entrypoints": {
    "shulkerboxtooltip": [
      "com.example.mymod.MyPluginClass"
    ]
},
```
Where `com.example.mymod.MyPluginClass` is the path the class that implements the interface.
From there you can register a custom preview provider to add support for you mod's boxes.
For documentation, see the javadocs in the `api` package.

If you have more question feel free to ask them ^^

**Comment by MisterPeModder** [comment]
My apoligies, I didn't interpret what you said correctly.
You don't have to do anything on your part, the fix should be easy to implement.

**Comment by MisterPeModder** [comment]
I tried to get the size of inventories using the last slot index like you suggested, but this means that users can set arbitrary sizes to their shulker boxes by just modifying the NBT data...
That would be quite the chaos if someone were to people with the mod installed themselves a shulker box with Integer.MAX_VALUE slots!

So instead I added the possiblity to overwrite preview providers in the API, that allows your mod to overwrite to the default shulker box preview.
I sent you a pull request that implements this feature.
